### DISCUSSION TITLE
Relay serve secret configs

### DISCUSSION THREAD

**Discussion by arvindk-dev** [practable/relay#54] [Q&A] [answered]
Hello,

I'm trying to setup the relay application in my machine. While executing 'relay serve' command I'm prompted for RELAY_SECRET which is a .pat file.

Please provide more information about this .pat file and help me in generating it so that I can set up the relay application.

I'm new to Go lang, so it would be helpful if you can share some additional documentation for setting up, building and using this relay application.

**Comment by timdrysdale** [comment, marked as answer]
Hi I'm delighted you are giving this a go. Yes, documentation is still pretty minimal at this stage. I appreciate it is a chicken and egg situation, but if you get far enough in, we'd welcome a hand with that. Meanwhile, happy to receive Q&A.
The content of the pat file can be any string. It is not formatted in any special way. It's just the "secret" string that is used to sign the JWT tokens. Relay needs to know what it is so it can check that the JWT tokens it gets from clients are signed by a trusted provider of JWT tokens (defined as anyone who knows the secret). Typically the management layer will be signing tokens for clients.

**Comment by arvindk-dev** [comment]
Hi Tim,
Thanks for your timely response and providing the details.
We'll try to explore more and integrate relay to our application.
